### PR TITLE
dynamic pagination

### PR DESCRIPTION
**Pull request by annashamray**
fixes https://github.com/open-zaak/open-zaak/issues/1668 (common ground part)

**Changes:**
* add dynamic pagination class and mixin
* drop support for python 3.8 and 3.9

### CODE REVIEW

File: tests/test_pagination.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+@pytest.mark.django_db
+def test_list_with_page_size_in_query(api_client):
+    HobbyFactory.create_batch(10)
+    path = reverse("paginate-hobby-list")
+
+    response = api_client.get(path, {"pageSize": 5})
+    assert response.status_code == status.HTTP_200_OK
+
+    data = response.json()
+
```
Comment (SonnyBA) [inline tests/test_pagination.py:32]: I would expect here to also assert that the amount of returned items equals that of the given `pageSize`

File: vng_api_common/pagination.py [hunk 2]
```diff
@@ -0,0 +1,10 @@
+from rest_framework.pagination import PageNumberPagination
+
+
+class DynamicPageSizeMixin:
+    page_size = 100
+    page_size_query_param = "pageSize"
+    max_page_size = 500
+
+
+class DynamicPageSizePagination(DynamicPageSizeMixin, PageNumberPagination): ...
```
Comment (SonnyBA) [inline vng_api_common/pagination.py:10]:
```suggestion
class DynamicPageSizePagination(DynamicPageSizeMixin, PageNumberPagination):
    page_size = 100
    page_size_query_param = "pageSize"
    max_page_size = 500
```

File: vng_api_common/pagination.py [hunk 3]
```diff
@@ -0,0 +1,10 @@
+from rest_framework.pagination import PageNumberPagination
+
+
+class DynamicPageSizeMixin:
+    page_size = 100
+    page_size_query_param = "pageSize"
+    max_page_size = 500
```
Comment (SonnyBA) [inline vng_api_common/pagination.py:7]:
```suggestion
```

**Review by SonnyBA — CHANGES_REQUESTED** [review]
I think asserting that the correct amount of items are returned in the responses would be nice to add to the tests.

File: vng_api_common/pagination.py (hunk 3 above)
Comment (annashamray) [reply vng_api_common/pagination.py:7]: I don't quite understand this suggestion. Do you want the mixin to be removed? 
In this way It won't be used for custom pagination classes (like in Open Zaak)

Comment (SonnyBA) [reply vng_api_common/pagination.py:7]: I had not thought about re-using the mixin for other projects so you can ignore the suggestion :+1:

**Review by SonnyBA — APPROVED** [review]
Looks good :+1:

**Pull request merged**